icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -642,9 +613,10 @@ SimpleDateFormat& SimpleDateFormat::operator=(const SimpleDateFormat& other)
     }
 
     UErrorCode localStatus = U_ZERO_ERROR;
-    freeFastNumberFormatters();
-    initFastNumberFormatters(localStatus);
-
+    // if it's null, we should not reinit since it might be using an adopted number formatter
+    if (other.fSimpleNumberFormatter) {
+        initSimpleNumberFormatter(fLocale, localStatus);
+    }
```

Comment:
Nit: What you _should_ be doing here is unconditionally calling the copy constructor on fSimpleNumberFormatter. But since SimpleNumberFormatter doesn't have a copy constructor, you need to do this special condition thing. Please update the comment to state that.

Alternatively, and maybe better, unconditionally call initSimpleNumberFormatter, since you now brand-check for DecimalFormat inside it.